### PR TITLE
feat(api): add friend request api endpoints

### PR DESCRIPTION
**Pull request by tgrigoruk**
### Summary
Add API endpoints for for creating, updating, and getting friend requests.

- [x] GET `/api/v1/friendship/requests`
- [x] POST `/api/v1/friendship/:userId`
- [x] PATCH `/api/v1/friendship/:userId`
- [x] add requisite dummy data for testing

### Retrospective
Reflection of things learned or hardships encountered -- used as learning resource

### Testing Notes
Trying to keep it simple so I'm not handling various cases, and am completely ignoring `blocked` status (perhaps we should even remove it).

### Testing
Test using Postman etc.
1. POST `http://localhost:8000/api/v1/auth/login` with body `{"email":"alice.smith@email.com", "password":"password"}`
2. GET `http://0.0.0.0:3000/api/v1/user/friends` shows current friends (no John)
3. GET `http://0.0.0.0:3000/api/v1/friendship/requests` will show request from John
4. PATCH `http://0.0.0.0:3000/api/v1/friendship/2` with body `{"status":"accepted"}` will return with `"updated": true`
5. step 2 again to see John added
6. step 3 again returns empty list
7. POST `http://0.0.0.0:3000/api/v1/friendship/1` with body `{"message":"Up for a game of bAdminton?"}` responds with `"created": true` only on first time sent. See new friendship in db.

### CODE REVIEW

File: apps/api/src/server/services/UserService.ts [hunk 1]
```diff
@@ -48,6 +54,92 @@ export class UserController {
...
+        instagram: users.instagram,
+      })
+      .from(users)
+      .where(inArray(users.id, friendIds)) as FriendProfile[];
+  }
+
+  public async getFriendslist(id: number) {
+    const friendIds = await this.getFriendshipIds(id);
+    const friendProfiles = await this.getFriendProfiles(friendIds);
+    const friendsOfFriends = await Promise.all(friendProfiles.map(friend => this.getFriendshipIds(friend.id!)));
```
Comment (tgrigoruk) [inline apps/api/src/server/services/UserService.ts:93]: If User had N friends and each have M friends then this is 1 + (N x M) db queries! Terrible, but it works. I don't think it's worth the effort to optimize this.

File: apps/api/src/server/services/UserService.ts [hunk 2]
```diff
@@ -48,6 +54,92 @@ export class UserController {
...
+      .select({
+        inviterId: friendships.inviterId,
+        inviteeId: friendships.inviteeId,
+        message: friendships.message,
+      })
+      .from(friendships)
+      .where(and(
+        eq(friendships.status, 'pending'),
+        eq(friendships.inviteeId, id),
+      ));
```
Comment (tgrigoruk) [inline apps/api/src/server/services/UserService.ts:114]: Would like to use `getFriendshipIds()` but this only looks for friendships where User is _invitee_.

File: apps/api/src/server/services/UserService.ts [hunk 3]
```diff
@@ -48,6 +54,96 @@ export class UserController {
...
+  }
+
+  public async updateFriendRequest(id:number, friendId: number, status:FriendshipStatus) {
+    const results = await this.resources.db
+      .update(friendships)
+      .set({ status })
+      .where(and(
+        eq(friendships.inviterId, friendId),
+        eq(friendships.inviteeId, id),
+        eq(friendships.status, 'pending'),
```
Comment (tgrigoruk) [inline apps/api/src/server/services/UserService.ts:142]: Only updates requests received from other users that have not been responded to yet.

File: apps/api/src/server/services/UserService.ts [hunk 4]
```diff
@@ -85,4 +181,63 @@ export class UserService {
...
+
+    if (inviteeId === inviterId) {
+      return new Error('You cannot send a friend request to yourself');
+    }
+
+    // TODO: different responses depending on the status of the friendship
+    const existingFriendship = await this.controller.getFriendshipIds(inviterId!, ['accepted', 'pending', 'blocked', 'rejected']);
+    if (existingFriendship.includes(inviteeId)) {
+      return new Error('You are already have a relationship with this user');
+    }
```
Comment (tgrigoruk) [inline apps/api/src/server/services/UserService.ts:223]: There are 8 kinds of relationships to consider (4 statuses, and user could be inviter or invitee). Ain't got time for that. I *think* this should suffice.

**Review by toco-t — APPROVED** [review]
LGTM!

**Pull request merged**